src/meta/src/rpc/server.rs

```diff
@@ -438,4 +444,59 @@ mod tests {
             leader_count
         );
     }
+
+    /// Creates `number_of_nodes` meta nodes
+    /// Deletes leader and or lease `number_of_nodes` times
+    /// After each deletion asserts that we have the correct number of leader nodes
+    #[tokio::test]
+    async fn test_fencing() {
+        let meta_port = 1600;
+        let compute_port = 1700;
+        let number_of_nodes = 4;
+        use crate::rpc::{META_CF_NAME, META_LEADER_KEY, META_LEASE_KEY};
+        use crate::storage::Transaction;
+
+        let meta_store = Arc::new(MemStore::default());
+        let vec_meta_handlers = setup_n_nodes_inner(number_of_nodes, meta_port, &meta_store).await;
+
+        // we should have 1 leader on startup
+        let leader_count = number_of_leaders(number_of_nodes, meta_port, compute_port).await;
+        assert_eq!(
+            leader_count, 1,
+            "Expected to have 1 leader at beginning, instead got {} leaders",
+            leader_count
+        );
+
+        let del = vec![(true, true), (true, false), (false, true)];
+
+        for i in 1..number_of_nodes {
+            let delete_leader = del[i as usize].0;
+            let delete_lease = del[i as usize].1;
```

Comment:
```suggestion
        for (delete_leader, delete_lease) in del {
```

Reply:
Thanks for pointing that out. Should have done that in the first place :D